### ISSUE TITLE
PSR2.Namespaces.UseDeclaration not giving error for use statements before the namespace declaration

### ISSUE THREAD

**Issue by artureiji** [squizlabs/PHP_CodeSniffer#1876]
The PSR2 Namespaces/UseDeclarationSniff does not seem to be validating the `When present, all use declarations MUST go after the namespace declaration.` rule properly.

Example (passes without any errors!):

```
<?php

use DateTime;
use Exception;
use Throwable;

namespace App;

class Foo
{
    protected $bar;
}

```

phpcs --version
PHP_CodeSniffer version 3.2.2 (stable) by Squiz (http://www.squiz.net)

Running: phpcs -s --standard=PSR2 test.php

**Comment by artureiji** [comment]
Any updates on this?

**Comment by gsherwood** [comment]
No, I haven't got to this yet.

**Comment by gsherwood** [comment]
Thanks for reporting this. Fix will be in 3.3.0.

**Comment by artureiji** [comment]
Neat! Thanks for the support @gsherwood
